apps/agent/test/agent.js

```diff
@@ -0,0 +1,577 @@
+const Agent = artifacts.require('Agent')
+
```

Comment:
It looks like all the tests are going through the Agent contract. There should be dedicated tests for SignatureValidator contract that will test all its public functions.